diff a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,7 @@ INSTALL=install
 
 SCRIPTS=shit-apply-patch-script shit-merge-one-file-script shit-prune-script \
 	shit-poop-script shit-tag-script shit-resolve-script shit-whatchanged \
-	shit-deltafy-script
+	shit-deltafy-script shit-fetch-script
 
 PROG=   shit-update-index shit-diff-files shit-init-db shit-write-tree \
 	shit-read-tree shit-commit-tree shit-cat-file shit-fsck-cache \
diff a/shit-fetch-script b/shit-fetch-script
--- /dev/null
+++ b/shit-fetch-script
@@ -0,0 +1,41 @@
+#!/bin/sh
+#
+merge_repo=$1
+merge_name=${2:-HEAD}
+
+: ${shit_DIR=.shit}
+: ${shit_OBJECT_DIRECTORY="${SHA1_FILE_DIRECTORY-"$shit_DIR/objects"}"}
+
+download_one () {
+	# remote_path="$1" local_file="$2"
+	case "$1" in
+	http://*)
+		wget -q -O "$2" "$1" ;;
+	/*)
+		test -f "$1" && cat >"$2" "$1" ;;
+	*)
+		rsync -L "$1" "$2" ;;
+	esac
+}
+
+download_objects () {
+	# remote_repo="$1" head_sha1="$2"
+	case "$1" in
+	http://*)
+		shit-http-poop -a "$2" "$1/"
+		;;
+	/*)
+		shit-local-poop -l -a "$2" "$1/"
+		;;
+	*)
+		rsync -avz --ignore-existing \
+			"$1/objects/." "$shit_OBJECT_DIRECTORY"/.
+		;;
+	esac
+}
+
+echo "Getting remote $merge_name"
+download_one "$merge_repo/$merge_name" "$shit_DIR"/MERGE_HEAD
+
+echo "Getting object database"
+download_objects "$merge_repo" "$(cat "$shit_DIR"/MERGE_HEAD)"
diff a/shit-poop-script b/shit-poop-script
--- a/shit-poop-script
+++ b/shit-poop-script
@@ -6,39 +6,7 @@ merge_name=${2:-HEAD}
 : ${shit_DIR=.shit}
 : ${shit_OBJECT_DIRECTORY="${SHA1_FILE_DIRECTORY-"$shit_DIR/objects"}"}
 
-download_one () {
-	# remote_path="$1" local_file="$2"
-	case "$1" in
-	http://*)
-		wget -q -O "$2" "$1" ;;
-	/*)
-		test -f "$1" && cat >"$2" "$1" ;;
-	*)
-		rsync -L "$1" "$2" ;;
-	esac
-}
-
-download_objects () {
-	# remote_repo="$1" head_sha1="$2"
-	case "$1" in
-	http://*)
-		shit-http-poop -a "$2" "$1/"
-		;;
-	/*)
-		shit-local-poop -l -a "$2" "$1/"
-		;;
-	*)
-		rsync -avz --ignore-existing \
-			"$1/objects/." "$shit_OBJECT_DIRECTORY"/.
-		;;
-	esac
-}
-
-echo "Getting remote $merge_name"
-download_one "$merge_repo/$merge_name" "$shit_DIR"/MERGE_HEAD
-
-echo "Getting object database"
-download_objects "$merge_repo" "$(cat "$shit_DIR"/MERGE_HEAD)"
+shit-fetch-script "$merge_repo" "$merge_name"
 
 shit-resolve-script \
 	"$(cat "$shit_DIR"/HEAD)" \
